src/lib/containers/dataaccess/AccessRequirementSearchBox.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import AsyncSelect from 'react-select/async'
+import { SynapseClient } from '../../utils'
+import { useSynapseContext } from '../../utils/SynapseContext'
+
+export type AccessRequirementSearchBoxProps = {
+  onChange: (accessRequirementId?: string | undefined) => void
+}
+export default function AccessRequirementSearchBox(
+  props: AccessRequirementSearchBoxProps,
+) {
+  const { accessToken } = useSynapseContext()
+  async function _loadOptions(inputValue: string) {
+    const data = await SynapseClient.searchAccessRequirements(accessToken, {
+      nameContains: inputValue,
+    })
+
+    const options =
+      data?.results.map(item => ({
+        id: item.id.toString(),
+        value: item.id.toString(),
+        label: `${item.name} ${
+          item.name !== item.id.toString() ? `(${item.id})` : ''
+        }`,
+      })) ?? []
+
+    return options
+  }
+
+  return (
+    <AsyncSelect
+      defaultOptions
+      cacheOptions
+      isClearable
```

Comment:
Truthy defaultOptions just results in the request firing before typing, which is fine. It will pre-populate the options.

Truthy cacheOptions will cache the results based on the input query. Since name is the only filter we're using for this component, the cache is useful + valid.

Reply:
Hopefully we do not need to support multiple pages of results for ARs

Reply:
Not for this component, at this time